### PR TITLE
Feature/HMI Policy Table Update using vehicle modem 

### PR DESCRIPTION
**Pull request by LitvinenkoIra**
Implements [HMI Policy Table Update using vehicle modem](https://github.com/smartdevicelink/sdl_core/issues/3161)

This PR is **[ready]** for review.

### Risk
This PR makes **[no]** API changes.

### Testing Plan
ATF test scripts: TBD

### Summary

-  Send UPDATING state after receiving a successful SDL.PolicyUpdate response

- Restart the timeout in case SDL received BC.OnSystemRequest with type PROPRIETARY or HTTP from HMI in UPDATING state

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by LitvinenkoIra** [comment]
@yang1070 please approve these changes

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/sdl_policy_update_response.cc [hunk 1]
```diff
@@ -52,7 +53,17 @@ SDLPolicyUpdateResponse::~SDLPolicyUpdateResponse() {}
 
 void SDLPolicyUpdateResponse::Run() {
   LOG4CXX_AUTO_TRACE(logger_);
-  // TODO(PV): add some logic here
+  const hmi_apis::Common_Result::eType code =
+      static_cast<hmi_apis::Common_Result::eType>(
+          (*message_)[strings::params][hmi_response::code].asInt());
+
+  if (hmi_apis::Common_Result::SUCCESS != code &&
```
[the remark this answers is not in the document]
Comment (LitvinenkoIra) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/sdl_policy_update_response.cc]: @ShobhitAd Thank you, fixed in 85c83cb06112582aad4c5bbeeb35ec7517f41850

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/on_system_request_notification.cc [hunk 2]
```diff
@@ -121,16 +121,15 @@ void OnSystemRequestNotification::Run() {
...
 
-#ifdef EXTERNAL_PROPRIETARY_MODE
   using namespace rpc::policy_table_interface_base;
   const auto request_type =
       static_cast<rpc::policy_table_interface_base::RequestType>(
           (*message_)[strings::msg_params][strings::request_type].asUInt());
 
-  if (RequestType::RT_PROPRIETARY == request_type) {
-    policy_handler_.ptu_retry_handler().OnSystemRequestReceived();
+  if (RequestType::RT_PROPRIETARY == request_type ||
```
[the remark this answers is not in the document]
Comment (LitvinenkoIra) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/on_system_request_notification.cc]: @ShobhitAd Thank you, fixed in bc7b05ddecaaf07e49bd4dfbf023d4a4c5f2d4e9

**Comment by LitvinenkoIra** [comment]
@ShobhitAd Please, let me know when you finish your review and I will squash commits and resolve conflicts, thank you

**Comment by ShobhitAd** [comment]
@LitvinenkoIra Approved the changes for the PR. I would also like to add a note for reference before you resolve the merge conflicts. We would appreciate if developers did not force push to a pr that is in review. It makes it difficult to review a PR because we cannot tell what was changed since the last review. In general, please do not rebase, do not force push. Instead, you can merge the current develop branch into your PR branch. Thank you

**Comment by LitvinenkoIra** [comment]
@ShobhitAd Thanks for the review, I merged the develop branch into the feature branch to resolve the conflicts in cf66f18a5271d8527f110a034f0b7a54e5fe9546

**Comment by LitvinenkoIra** [comment]
@ShobhitAd Please don't merge this PR right now, we performed regression testing after merging the develop branch into the feature branch and faced with one issue related to the PTU retry sequence functionality for external policy flow. I would say that it will take us one day to investigate the issue and perform regression testing one more time. I can notify you here in the comments or via Slack when all tests have passed (approximately 04.02 Tuesday morning) Is it okay?

**Comment by LitvinenkoIra** [comment]
@ShobhitAd Thanks for the review, I merged the develop branch into the feature branch to resolve the conflicts in 1cf403d 
We need some time to finish our regression testing, I will notify you when all tests have been passed and this PR will be ready to merge

**Comment by GetmanetsIrina** [comment]
@ShobhitAd, all conflicts are resolved, all regression scripts are passed.
PR is ready to merge.

**Pull request merged**